docker.go

```diff
@@ -46,9 +46,48 @@ type Parameters struct {
 
 // Volume is the docker volume details associated to the container
 type Volume struct {
-	ContainerPath string `json:"containerPath,omitempty"`
-	HostPath      string `json:"hostPath,omitempty"`
-	Mode          string `json:"mode,omitempty"`
+	ContainerPath string            `json:"containerPath,omitempty"`
+	HostPath      string            `json:"hostPath,omitempty"`
+	Mode          string            `json:"mode,omitempty"`
+	Persistent    *PersistentVolume `json:"persistent,omitempty"`
+}
+
+// PeristentVolume declares a Volume to be Persistent, and sets
+// the size (in MiBs) and optional constraints for the Volume.
+type PersistentVolume struct {
+	Type        *string     `json:"type,omitempty"`
```

Comment:
As there are only the three available types `root`, `path`, and `mount`, how about creating a string type declaration `PersistentVolumeType` and define three constants accordingly? This would make it easier for users to understand the valid input domain and also enable IDE/editor auto-completion.

(FWIW, Kubernetes does this a lot, like [here](https://github.com/kubernetes/kubernetes/blob/5f2846eefb48cf3ec1d1130c64a7007e19430200/pkg/api/v1/types.go#L495-L502)).

I also think we don't need a pointer as the empty string is not a valid option.